### DISCUSSION TITLE
DateTimeRange Filter

### DISCUSSION THREAD

**Discussion by DMXLABDev** [rappasoft/laravel-livewire-tables#1062] [Q&A]
Hi, 
great component. 

Is it possible to create a filter with a date and datetime range picker ? 

Thanks
Dario

**Comment by lrljoe** [comment]
It is indeed possible.  I created a DateRange filter using Flatpickr, I doubt it'll make it into core as I don't want @rappasoft to have to introduce dependencies on 3rd party libraries, but I'm happy to publish a version and drop a link in here if you'd like?

**Reply by DMXLABDev** [reply]
If you put a link here it would make me very happy. 
Also about the other plugins you pointed to dgillier.

Thanks

**Reply by lrljoe** [reply]
As I'm still slacking on finishing the documentation, here's a quick preview with most of the filters working across Tailwind 2/3, Bootstrap 4/5.
http://tabledemo.lowerrocklabs.com/
I've yet to finish tidying things up/making things look nicer, but gives you an idea of functionality.

**Reply by lrljoe** [reply]
Available here:
https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters

**Comment by dgillier** [comment]
Maybe it(s time to introduce some kind of plugins for filters ?
Select2, FlatPickr, etc...

**Reply by lrljoe** [reply]
I'm more than happy to publish the various filters I've built as extra packages.  What I have so far/ready to go is:
DateRange (Earliest and Latest Date) using flatpickr
Numeric Range (Min and Max numbers) using alpine
Select2 (i.e. search and select) variant using alpine
GeoLocation with Range ( from point) using alpine, GoogleMapsAPI

But I think this would really needs @rappasoft to give an indication of comfort with people adding plugins/extra filters outside of core, as I don't want to introduce pain if he's got things pending.

**Reply by lrljoe** [reply]
Just to note that these are published on a separate package linked below to avoid introducing dependencies. Depending on uptake I'll look at which I can more smoothly introduce into core.
[https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters](https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters)

**Comment by lrljoe** [comment]
Add-on package with these filters should be ready for some wider testing on Monday

**Comment by lrljoe** [comment]
Quick update, it's all mostly there, I'm just doing some final test creation/testing and verifying documentation actually matches reality.

**Reply by BenOussama180** [reply]
looking forward to test it !! @lrljoe

**Reply by lrljoe** [reply]
As I'm still slacking on finishing the documentation, here's a quick preview with most of the filters working across Tailwind 2/3, Bootstrap 4/5.
http://tabledemo.lowerrocklabs.com/
I've yet to finish tidying things up/making things look nicer, but gives you an idea of functionality.

DateTime should be the filter below the number range

**Comment by lrljoe** [comment]
As I'm still slacking on finishing the documentation, here's a quick preview with most of the filters working across Tailwind 2/3, Bootstrap 4/5.
[http://tabledemo.lowerrocklabs.com/](http://tabledemo.lowerrocklabs.com/)
I've yet to finish tidying things up/making things look nicer, but gives you an idea of functionality.

DateTime should be the filter below the number range

**Reply by lrljoe** [reply]
No longer slacking quite so much
Available here:
[https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters](https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters)

**Comment by lrljoe** [comment]
This is now available in the Core Package as of v3
